README.md

```diff
@@ -45,157 +53,294 @@ console.log(result);
+**Note**: the `createRecord` function performs no error checking and assumes all its inputs are correct. Incorrect inputs will have unpredictable consequences.
 
-const router = (request, response) => {
-    if (request.method == 'POST') {
-        let body = '';
+## Other Requirements
+* The server should not crash because of a client request.
+* If the client supplies bad data, the server should respond with a `400` (Bad request) status code and a helpful message indicating the problem.
 
-        request.on('data', (data) => {
-            body += data;
-        });
+## Steps
+1. Decide how your validation code will indicate that there is a problem with the payload. You are free to use any of the three approaches outlined above (or use another approach if you feel it is appropriate).
+2. Start by writing individual functions to validate each aspect of the payload. This will make it easier to adhere to TDD as you write them (which you should). We've started you off with `modules/validateFilename.js` and `tests/modules/validateFilename.test.js`.
+3. After you have written your individual functions, you can move on to completing the handler in `modules/handler.js`. This can also be approached using TDD by using the [`shot`](https://github.com/hapijs/shot) module to test your handler. We've started you off again with `modules/handler.js` and `tests/modules/index.test.js`.
+4. After you think you have finished, run the acceptance tests with `npm run acceptance`. If any of them fail, you have missed an edge case. Otherwise, you are done :tada:
 
-        request.on('end', () => {
-            const post = qs.parse(body);
-            // now deal with post
-        });
-    }
-}
-```
-Your router should call a handler which utilises the *validateName*, *validateAge*, *validateEmail*, *validateReason* to check the form contents, then write either a simple success or fail response to the user. On a related note, check out [this](https://www.smashingmagazine.com/2009/01/404-error-pages-one-more-time/) great article on handling 404 errors. You can test your router using the [shot](https://github.com/hapijs/shot) module, which is added as a dependency to this project. Good luck! :)
+### Notes
+These notes are important to be aware of in general, but are not necessary for the purposes of the workshop.
 
+#### Error-First Callback Pattern
+There are a couple of gotchas when using this callback pattern:
+* You _must_ ensure that the callback is not called more than once in your function. This can be done either using `if/else` blocks, `switch` statements (with `break`), or early `return` statements (e.g. `return callback(null, result)`).
+* When presenting a callback interface to the caller, it will expect the callback to be executed asynchronously. On Node.js, you can use [`process.nextTick`](https://nodejs.org/api/process.html#process_process_nexttick_callback_args) for this purpose, on the client-side, you can use `setTimeout(Function, 0)`.
```

Comment:
I think this bit will be confusing for some, but as it's only some extra info then it seems fine to leave it in